File: src/FibonacciN.java
import java.util.Scanner;

public class FibonacciN {
    void series(int cur,int prev,int series){
        if (prev > series ){
            return;
        }
        else {
            System.out.print(prev+" ");
            int temp = prev;
            prev = cur + prev;
            cur = temp;
            series(cur,prev,series);
        }
    }
    public static void main(String args[]){
        FibonacciN fibo = new FibonacciN();
        Scanner scn  = new Scanner(System.in);
        int testCase = scn.nextInt();
        while (testCase-- > 0) {
            int series = scn.nextInt();
            int cur = 0,next = 1;
            System.out.print(cur+" ");
//            System.out.print(next);
            fibo.series(cur,next,series);
        }
    }
}

File: src/FibonacciN.kt
class FibonacciN1{
    fun series(cur:Int,prev:Int,size:Int){
        var cur = cur
        var prev = prev
        var series = size
        if (prev > size){
            return
        }else{
            print(prev)
            print(" ")
            var temp = prev;
            prev = cur + prev;
            cur = temp;
            series(cur,prev,series);
        }
    }
}
fun  main(args : Array<String>){
    var testCase = readLine()!!.toInt()
    var cur = 0
    var prev = 1
    var fiboObje = FibonacciN1()
    while (testCase-- > 0){
        val seriesSize = readLine()!!.toInt()
        print(cur)
        print(" ")
        fiboObje.series(cur,prev,seriesSize)
        println()
    }
}